interactive_engine/executor/ir/core/src/glogue/pattern.rs

```diff
@@ -406,6 +384,19 @@ impl Pattern {
                             edge_expand,
                             &mut next_vertex_id,
                         )?;
+                        if let Some(BinderItem::Edge(_)) = binder.item.as_ref() {
+                            if edge_expand.expand_opt == pb::edge_expand::ExpandOpt::Edge as i32 {
+                                edge_expands_to_close.insert(dst_vertex_id, edge_direction);
+                            }
+                        } else if let Some(BinderItem::Path(_)) = binder.item.as_ref() {
+                            edge_expands_to_close.insert(dst_vertex_id, edge_direction);
+                        }
+                        if let Some(params) = vertex_params {
```

Comment:
Some() here doesn't mean vertex_params is queryable (e.g., has predicates). By default, it is a Some(Params::default()).

Reply:
implement is_empty method for params to check whether we need to add it